### ISSUE TITLE
Site Editor: `RichText` component in a block does not change cursor style

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#53374]
### Description

This may depend on the operating system, but in Windows OS, the mouse cursor behaves as follows.

- By default, it is an arrow.
- Changes to the text shape when the mouse hovers over an editable field
- The cursor blinks when the mouse hovers over an editable field

These behaviors allow the user to see if the field is editable and where the cursor is focusing. In the Site Editor, however, the cursor remains the default arrow when you mouse over `RichText` in a block. Also, clicking on the field does not show a blinking cursor, so the user cannot see that the field has the focus.

This problem does not occur in the submission editor. Perhaps it is due to the fact that in the Site Editor, the cursor style is being overridden by the following style.

![cursor-default](https://github.com/WordPress/gutenberg/assets/54422211/58f16438-4767-4681-bb88-dec31b403af9)

### Step-by-step reproduction instructions

- Access to the Site Editor.
- Add a table/image/gallery block (These blocks have a `RichText` component as caption)
- Mouse over the caption area.
- The mouse cursor style should not change.
- Click on the caption area.
- A blinking cursor should not appear in the caption area.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/54422211/a125890f-913c-40b4-9adb-9ed3dce178c4

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
